pymc/backends/ndarray.py

```diff
@@ -115,6 +115,8 @@ def record(self, point, sampler_stats=None) -> None:
         if sampler_stats is not None:
             for data, vars in zip(self._stats, sampler_stats):
                 for key, val in vars.items():
+                    if isinstance(val, np.ndarray) and val.shape[0] == 1:
```

Comment:
Can val have more than one dim? More importantly it would be good if we didn't need this check in every iteration/ for every stat. 

Can we solve it from another angle?

Reply:
none of the tests passed in any values of greater than one dimension -- but noted on the point about trying to solve this outside the loop!